packages/fuel-gauge/src/predicate/predicate-estimations.test.ts

```diff
@@ -204,5 +209,50 @@ describe('Predicate', () => {
       expect(estimatePredicatesSpy).toHaveBeenCalledTimes(1);
       expect(dryRunSpy).toHaveBeenCalledOnce();
     });
+
+    describe('should ensure resources inputs from a predicate includes predicateData', () => {
+      test('fetching with predicate', async () => {
+        const transactionRequest = new ScriptTransactionRequest();
+
+        const resources = await predicateStruct.getResourcesToSpend([[fundingAmount, baseAssetId]]);
+        resources.forEach((resource) => {
+          expect(resource.predicateData).toBeDefined();
+        });
+
+        transactionRequest.addResources(resources);
+        const inputs = transactionRequest.inputs.filter(isRequestInputResource);
+
+        expect(inputs.length).toBeGreaterThan(0);
+        inputs.forEach((resource) => {
+          expect(resource.predicateData).toBeDefined();
+        });
+      });
+
+      test('fetching with provider', async () => {
```

Comment:
Based on the `describe` above titled `should ensure resources inputs from a predicate includes predicateData`, this test `fetching with provider` implies that fetching predicate resources with `provider.getResourcesToSpend` populates the `predicateData`, which doesn't match the contents of the test.

I suggest renaming the `describe` and `test`s, e.g.
- describe: `predicate resource fetching and predicateData population`,
    - t1: `getting predicate resources via the predicate automatically populates predicateData`
    - t2: `getting predicate resources via the provider requires manual predicateData population`

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2380/commits/4179a9cada578e04ba165b60adda817dd5afa659